bcl2fastq/index_summary.py

```diff
@@ -5,66 +5,110 @@
+
+
+#           Dropdown option's label set to all_runs(combination of unknown and known)
+#           Dropdown multi set to false (default) - only single select possible
+#                 -  multiple select gives error due to inconsistent values
 layout = html.Div(children=[
     dcc.Dropdown(
         id='run_select',
         options=all_runs,
         value=all_runs[0]['value'],
+        clearable=False
     ),
+
     dcc.Graph(
         id='known_index_bar',
+
     ),
+    # Bar graph with defaults set, no change in colour etc
     html.Div([
         html.Div(
             [dcc.Graph(id='known_unknown_pie'),
              dcc.Textarea(
                  id='known_fraction',
                  style={'width': '100%'},
                  readOnly=True,
+                 # This is the textbox at the bottom, hover over to see title
                  title=(
                      "Assumptions are made about which indexes are known "
                      "or unknown. This is due to multiple bcl2fastq analyses "
                      "being used on one run. This number should be 100%."
                  ),
+
              )],
-            style={'width': '24%', 'display': 'inline-block'}
+            style={'width': '24%', 'display': 'inline-block', }
         ),
         html.Div(
             [dcc.Graph(id='unknown_index_bar')],
-            style={'width': '74%', 'display': 'inline-block', 'float': 'right'}
+            style={'width': '76%', 'display': 'inline-block', 'float': 'center'}
         ),
     ]),
+    #           Sample_run_hidden holds json split format of "Known"
+    #               columns: "FlowCell","Index1","Index2","LIMS IUS SWID","LaneClusterPF","LaneClusterRaw",
+    #                   "LaneNumber","LaneYield","QualityScoreSum","ReadNumber","Run","RunNumber","SampleID",
+    #                   "SampleName","SampleNumberReads","SampleYield","TrimmedBases","Yield","YieldQ30"
+    #               Index: 6761 - 8307
+
+    #           pruned_unknown_hidden holds json split format of "unknown"
+    #               columns:  "Count","LaneNumber","Index1","Index2","Run","LIMS IUS SWID"
+    #               Index: 218463 - 227804
+
     html.Div(id='sample_run_hidden', style={'display': 'none'}),
     html.Div(id='pruned_unknown_hidden', style={'display': 'none'}),
+
 ])
 
 try:
     from app import app
 except ModuleNotFoundError:
     import dash
+
     app = dash.Dash(__name__)
     app.layout = layout
 
 
 @app.callback(
+
     dep.Output('sample_run_hidden', 'children'),
     [dep.Input('run_select', 'value')]
+
 )
+#           Replace run variable iteratively
 def update_sample_run_hidden(run_alias):
+    """ When input (run_select) is changed, function is rerun to get json data for selected run
+
+        Attributes:
+            run(list): run number
+
+        Parameters:
+            run_alias(list): alias of the run
+
+        Returns:
+             json object with columns, indices, and data, from run variable
```

Comment:
In the `Returns` field, also specify what this function changes. In this case, the hidden div JSON string is updated. This applies to the subsequent annotations as well.

Reply:
I would have put that at the top in the description, not at the bottom. From what I can see "updating the hidden div JSON string" is a side effect of this method, and not actually part of the return value, and there's no indication in the code itself (other than the name of the function) that this happens.

Reply:
The `@app.callback` decorator just above the function definition makes these connections explicit.

The function parameters are values in the Dash Layout that have just changed and the return value will become the new value in the Dash Layout.

@ijafri26 I will be happy with either convention you pick (in function description or part of Returns description), but be consistent.

Reply:
Sure, that makes sense to me. @ijafri26, could you please tweak some of the `Returns` descriptions? Instead of `updates json string "sample_run_hidden"`, which says what the code does, can you describe the effect that this has on the page? You already do this for some of them below, like `updates bar graph "known_index_bar"`.